ams/server/src/main/java/com/netease/arctic/server/table/executor/DataExpiringExecutor.java

```diff
@@ -0,0 +1,449 @@
+
+    if (table.isKeyedTable()) {
+      KeyedTable keyedTable = table.asKeyedTable();
+      ChangeTable changeTable = keyedTable.changeTable();
+      BaseTable baseTable = keyedTable.baseTable();
+      Snapshot changeSnapshot = changeTable.currentSnapshot();
+      Snapshot baseSnapshot = baseTable.currentSnapshot();
+
+      TableEntriesScan changeScan = fileEntriesScan(changeTable, partitionFilter, changeSnapshot);
+      TableEntriesScan baseScan = fileEntriesScan(baseTable, partitionFilter, baseSnapshot);
+      ExpireFiles changeExpiredFiles = new ExpireFiles();
+      ExpireFiles baseExpiredFiles = new ExpireFiles();
+      CloseableIterable<FileEntry> changed = CloseableIterable.transform(changeScan.entries(),
+          e -> new FileEntry(e, true));
+      CloseableIterable<FileEntry> based = CloseableIterable.transform(baseScan.entries(),
+          e -> new FileEntry(e, false));
+
+      try (CloseableIterable<FileEntry> entries = CloseableIterable.withNoopClose(Iterables.concat(changed, based))) {
+        CloseableIterable<FileEntry> mayExpiredFiles = CloseableIterable.withNoopClose(
+            Lists.newArrayList(CloseableIterable.filter(entries,
+                e -> mayExpired(e, expirationConfig, partitionFreshness, expireTimestamp))));
+        CloseableIterable.filter(mayExpiredFiles, e -> willNotRetain(e, expirationConfig, partitionFreshness))
+            .forEach(e -> {
+              if (e.isChange) {
+                changeExpiredFiles.addFile(e);
+              } else {
+                baseExpiredFiles.addFile(e);
+              }
+            });
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      expireFiles(changeTable, changeSnapshot.snapshotId(), changeExpiredFiles, expireTimestamp);
+      expireFiles(baseTable, baseSnapshot.snapshotId(), baseExpiredFiles, expireTimestamp);
+    } else {
+      UnkeyedTable unkeyedTable = table.asUnkeyedTable();
+      Snapshot snapshot = unkeyedTable.currentSnapshot();
+      ExpireFiles expiredFiles = new ExpireFiles();
+      try (CloseableIterable<IcebergFileEntry> entries
+          = fileEntriesScan(unkeyedTable, partitionFilter, snapshot).entries()) {
+        CloseableIterable<IcebergFileEntry> mayExpiredFiles = CloseableIterable.withNoopClose(
+            Lists.newArrayList(CloseableIterable.filter(entries,
+                e -> mayExpired(e, expirationConfig, partitionFreshness, expireTimestamp))));
+        CloseableIterable.filter(
+                mayExpiredFiles,
+                e -> willNotRetain(e, expirationConfig, partitionFreshness))
+            .forEach(expiredFiles::addFile);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      expireFiles(unkeyedTable, snapshot.snapshotId(), expiredFiles, expireTimestamp);
+    }
+  }
+
+  private static Expression getPartitionExpression(DataExpirationConfig expirationConfig, long expireTimestamp) {
```

Comment:
How about we generate Expression through `org.apache.iceberg.expressions.Literal`, which has a lot of conversion logic built in. WDYT?